tests/integration/istioio/trafficmanagement/circuitbreaker/scripts/trip_circuitbreaker.txt

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# $snippet test_fortio_httpbin_interaction.sh syntax="bash"
+$ FORTIO_POD=$(kubectl get pods -n istio-io-circuitbreaker | grep fortio | awk '{ print $1 }')
+$ kubectl -n istio-io-circuitbreaker exec -it $FORTIO_POD -c fortio /usr/bin/fortio -- load -curl http://httpbin:8000/get
+# $verify verifier="contains"
+200 OK
+# $endsnippet
+
+# $snippet almost_trip_circuit_breaker.sh syntax="bash"
+$ kubectl -n istio-io-circuitbreaker exec -it $FORTIO_POD -c fortio /usr/bin/fortio -- load -c 2 -qps 0 -n 20 -loglevel warning http://httpbin:8000/get
+# $endsnippet
+
+# $snippet trip_circuit_breaker.sh syntax="bash"
+$ kubectl -n istio-io-circuitbreaker exec -it $FORTIO_POD -c fortio /usr/bin/fortio -- load -c 3 -qps 0 -n 30 -loglevel warning http://httpbin:8000/get
```

Comment:
There needs to be verification that some of these requests failed with a 503. The other steps need verification as well.

Reply:
This can be used as an example of how to compare the percentage. Not the tolerance sinc the number of 503s returned is not exact. 

https://github.com/istio/istio/blob/master/tests/integration/istioio/trafficmanagement/scripts/traffic_shifting.txt#L42..L68

Reply:
I added verification of percentage ranges. 

I also wanted to get the output as well by specifying `outputis=` (but removed now) to match what is there in istio.io site. But for some reason, it did not produce any `.stdout.txt` or `.stdin.txt` files as output for `almost_trip_circuit_breaker.sh`, `trip_circuit_breaker.sh` and `print_statistics_after_tripping.sh` snippets in my local environment. 

I dug into test framework's code to see why it happens and found the following
```go
func (i snippetInfo) needsOutput() bool {
	return len(i.commandLines) > 0 && (i.outputIs != "" || len(i.verifiers) > 0)
}
```
My snippet has a command and even when I specify `outputis=`, stdout and stderr output files are not created. Here is the command generated when `outputis=` is specified
```bash
kubectl -n istio-io-circuitbreaker exec -i $FORTIO_POD -c fortio /usr/bin/fortio -- \
    load -c 2 -qps 0 -n 20 -loglevel warning http://httpbin:8000/get 1> >(tee /home/suchith/workspace/go/src/istio.io/istio/out/linux_amd64/trafficmanagement-ae642f7eaba54/TestCircuitBreaker/almost_trip_circuit_breaker.sh.stdout.txt) 2> >(tee /home/suchith/workspace/go/src/istio.io/istio/out/linux_amd64/trafficmanagement-ae642f7eaba54/TestCircuitBreaker/almost_trip_circuit_breaker.sh.stderr.txt >&2)


kubectl -n istio-io-circuitbreaker exec -i $FORTIO_POD -c fortio /usr/bin/fortio -- \
    load -c 3 -qps 0 -n 30 -loglevel warning http://httpbin:8000/get 1> >(tee /home/suchith/workspace/go/src/istio.io/istio/out/linux_amd64/trafficmanagement-ae642f7eaba54/TestCircuitBreaker/trip_circuit_breaker.sh.stdout.txt) 2> >(tee /home/suchith/workspace/go/src/istio.io/istio/out/linux_amd64/trafficmanagement-ae642f7eaba54/TestCircuitBreaker/trip_circuit_breaker.sh.stderr.txt >&2)
```
I then removed it and piped the output to a bash function I wrote (the present state)

Reply:
Ok! I figured that out. I had to run docker containers without `-it` as it caused issues. So the output was not piped to stdout. I had to add `2>&1` to the end so that the output is piped to stdout. Then things went fine.

Reply:
@brian-avery - While verifying percentages I have set 80-100% status 200 and 0-20% status 503 when load is less (almost tripped section where 20 requests are sent). For actual circuit breaking test, I have set it for 25-75% for both 200 and 503. Please feel free to suggest the percentage ranges for verification as setting this wrong could make the test flaky.